qiskit/quantum_info/mitigation/complete_readout_mitigator.py

```diff
@@ -0,0 +1,275 @@
+        self._assignment_mat = amat
+        self._mitigation_mats = {}
+
+    def expectation_value(
+        self,
+        data: Counts,
+        diagonal: Callable,
+        qubits: Iterable[int] = None,
+        clbits: Optional[List[int]] = None,
+        shots: Optional[int] = None,
+    ) -> Tuple[float, float]:
+        r"""Compute the mitigated expectation value of a diagonal observable.
+        This computes the mitigated estimator of
+        :math:`\langle O \rangle = \mbox{Tr}[\rho. O]` of a diagonal observable
+        :math:`O = \sum_{x\in\{0, 1\}^n} O(x)|x\rangle\!\langle x|`.
+        Args:
+            data: Counts object
+            diagonal: Optional, the vector of diagonal values for summing the
+                      expectation value. If ``None`` the the default value is
+                      :math:`[1, -1]^\otimes n`.
+            qubits: Optional, the measured physical qubits the count
+                    bitstrings correspond to. If None qubits are assumed to be
+                    :math:`[0, ..., n-1]`.
+            clbits: Optional, if not None marginalize counts to the specified bits.
+        Returns:
+            (float, float): the expectation value and standard deviation.
+        Raises:
+            QiskitError: if input arguments are invalid.
+        Additional Information:
+            The diagonal observable :math:`O` is input using the ``diagonal`` kwarg as
+            a list or Numpy array :math:`[O(0), ..., O(2^n -1)]`. If no diagonal is specified
+            the diagonal of the Pauli operator
+            :math`O = \mbox{diag}(Z^{\otimes n}) = [1, -1]^{\otimes n}` is used.
+            The ``clbits`` kwarg is used to marginalize the input counts dictionary
+            over the specified bit-values, and the ``qubits`` kwarg is used to specify
+            which physical qubits these bit-values correspond to as
+            ``circuit.measure(qubits, clbits)``.
+        """
+        # Marginalize counts
+        if clbits is not None:
+            data = marginal_counts(data, clbits)
+
+        # Get probability vector
+        num_qubits = data.num_qubits()
+        probs_vec = self._to_probs_vec(data, num_qubits)
+        shots = data.shots()
+
+        # Get qubit mitigation matrix and mitigate probs
+        if qubits is None:
+            qubits = range(num_qubits)
+        mit_mat = self.mitigation_matrix(qubits)
+
+        # Get operator coeffs
+        if diagonal is None:
+            diagonal = self._z_diagonal(2 ** num_qubits)
+        else:
+            diagonal = self._str2diag(diagonal)
+
+        # Apply transpose of mitigation matrix
+        coeffs = mit_mat.T.dot(diagonal)
+
+        return self._expval_with_stddev(coeffs, probs_vec, shots)
+
+    def quasi_probabilities(
+        self,
+        data: Counts,
+        qubits: Optional[List[int]] = None,
+        clbits: Optional[List[int]] = None,
+        num_qubits: Optional[int] = None,
```

Comment:
This also has different signature to base class method, clbits and num qubits.

Reply:
which of the options is better?